File: HandSock-Server/src/main/java/com/server/handsock/clients/service/ClientReportService.java
package com.server.handsock.clients.service;

import com.server.handsock.console.ConsolePrints;
import com.server.handsock.console.HandleResults;
import com.server.handsock.clients.dao.ClientChatDao;
import com.server.handsock.clients.dao.ClientReportDao;
import com.server.handsock.clients.man.ClientReportManage;
import com.server.handsock.clients.mod.ClientReportModel;
import lombok.Getter;
import lombok.Setter;
import org.springframework.stereotype.Service;

import java.util.Map;
import java.util.Objects;

@Service @Setter @Getter
public class ClientReportService {

    private final ClientChatDao clientChatDao;
    private final ClientReportDao clientReportDao;
    private final HandleResults handleResults = new HandleResults();

    public ClientReportService(ClientReportDao clientReportDao, ClientChatDao clientChatDao) {
        this.clientChatDao = clientChatDao;
        this.clientReportDao = clientReportDao;
    }

    public Map<String, Object> addReport(Map<String, Object> service, String sid, Long reporter_id, Long reported_id, String reason) {
        try {
            if (Objects.equals(reporter_id, reported_id)) {
                return handleResults.handleResultByCode(400, null, "不能举报自己");
            } else {
                ClientReportModel clientReportModel = new ClientReportModel();
                new ClientReportManage().insertRepo(clientReportModel, sid, reporter_id, reported_id, reason);
                if (clientReportDao.insert(clientReportModel) > 0) {
                    return handleResults.handleResultByCode(200, null, "举报成功");
                } else return handleResults.handleResultByCode(400, null, "举报失败");
            }
        } catch (Exception e) {
            new ConsolePrints().printErrorLog(e);
            return handleResults.handleResultByCode(500, null, "服务端异常");
        }
    }
}

File: HandSock-Server/src/main/java/com/server/handsock/clients/service/ClientReportService.kt
package com.server.handsock.clients.service

import com.server.handsock.clients.dao.ClientReportDao
import com.server.handsock.clients.man.ClientReportManage
import com.server.handsock.clients.mod.ClientReportModel
import com.server.handsock.utils.HandUtils
import com.server.handsock.utils.IDGenerator
import lombok.Getter
import lombok.Setter
import org.springframework.stereotype.Service

@Service @Setter @Getter
class ClientReportService(private val clientReportDao: ClientReportDao) {
    fun addReport(sid: String?, reporterId: Long, reportedId: Long, reason: String?): Map<String, Any> {
        try {
            if (reporterId == reportedId) {
                return HandUtils.handleResultByCode(400, null, "不能举报自己")
            } else {
                val clientReportModel = ClientReportModel()
                ClientReportManage(HandUtils, IDGenerator).insertRepo(clientReportModel, sid, reporterId, reportedId, reason)
                return if (clientReportDao.insert(clientReportModel) > 0) {
                    HandUtils.handleResultByCode(200, null, "举报成功")
                } else HandUtils.handleResultByCode(400, null, "举报失败")
            }
        } catch (e: Exception) {
            return HandUtils.printErrorLog(e)
        }
    }
}
